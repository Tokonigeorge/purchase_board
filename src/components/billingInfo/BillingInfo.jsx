import { useState } from "react";
import PaymentButton from "../PaymentButton";
import TextInput, { Label, textInputStyle } from "../TextInput";

const BillingInfo = ({ handleNav, handleCancel }) => {
  const [cardDetails, setCardDetails] = useState("");
  const [options, setOptions] = useState("Visa");

  const cards = ["Visa", "MasterCard", "Verve"];

  const validateDetails = (e) => {
    setCardDetails(e.target.value);
  };

  return (
    <div className="mt-10">
      <form>
        <TextInput
          label="Name on Card"
          isNecessary={true}
          placeholder="Opara Linus Ahmad"
        />
        <Label label={"Card Type"} isNecessary={true} />
        <select
          id="Card Type"
          className={textInputStyle}
          defaultValue={options}
          onChange={(e) => setOptions(e.target.value)}
        >
          {cards.map((i, indx) => (
            <option
              value={i}
              label={i}
              key={`${i + indx}`}
              className="py-2"
            ></option>
          ))}
        </select>
        <div className="flex-col flex sm:flex-row items-stretch w-full justify-between gap-x-12">
          <div className="w-full">
            <Label label={"Card Details"} isNecessary={true} />
            <input
              id="Card Details"
              type="text"
              inputMode="numeric"
              className={textInputStyle}
              value={cardDetails}
              maxLength={20}
              autoComplete="cc-number"
              onChange={(e) => validateDetails(e)}
            />
          </div>
          <div className="w-full sm:w-1/2 flex flex-row items-stretch gap-x-12">
            <div className="w-20">
              <Label label={"Expiry Date"} isNecessary={true} />
              <input
                id="Expiry Date"
                type={"text"}
                maxLength={5}
                placeholder="04/23"
                className={textInputStyle}
              />
            </div>
            <div className="w-12">
              <Label label={"CVV"} isNecessary={true} />
              <input
                id="CVV"
                type={"text"}
                maxLength={3}
                placeholder="923"
                className={`${textInputStyle} text-black `}
              />
            </div>
          </div>
        </div>
        <PaymentButton handleNav={handleNav} handleCancel={handleCancel} />
      </form>
    </div>
  );
};

export default BillingInfo;
