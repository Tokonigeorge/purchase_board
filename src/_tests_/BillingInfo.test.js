/**
 * @jest-environment jsdom
 */
import * as React from "react";

import renderer from "react-test-renderer";
import BillingInfo from "../components/billingInfo/BillingInfo";

describe("It renders Sucessfully", () => {
  test("eventHandler called on click", () => {
    const tree = renderer.create(<BillingInfo />);
    //create a snapshot for the component
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
