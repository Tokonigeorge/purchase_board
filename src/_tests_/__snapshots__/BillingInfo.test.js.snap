// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders Sucessfully eventHandler called on click 1`] = `
<div
  className="mt-10"
>
  <form>
    <label
      className="font-bold text-sm"
      htmlFor="Name on Card"
    >
      Name on Card
       
      <span
        className="text-midRed"
      >
        *
      </span>
    </label>
    <br />
    <input
      className="mt-2 w-full ring-1 rounded-md py-2.5 px-3 outline-none ring-purp mb-8 text-textgray text-sm"
      data-testid="text-input"
      id="Name on Card"
      onChange={[Function]}
      placeholder="Opara Linus Ahmad"
      type="text"
      value=""
    />
    <label
      className="font-bold text-sm"
      htmlFor="Card Type"
    >
      Card Type
       
      <span
        className="text-midRed"
      >
        *
      </span>
    </label>
    <br />
    <select
      className="mt-2 w-full ring-1 rounded-md py-2.5 px-3 outline-none ring-purp mb-8 text-textgray text-sm"
      defaultValue="Visa"
      id="Card Type"
      onChange={[Function]}
    >
      <option
        className="py-2"
        label="Visa"
        value="Visa"
      />
      <option
        className="py-2"
        label="MasterCard"
        value="MasterCard"
      />
      <option
        className="py-2"
        label="Verve"
        value="Verve"
      />
    </select>
    <div
      className="flex-col flex sm:flex-row items-stretch w-full justify-between gap-x-12"
    >
      <div
        className="w-full"
      >
        <label
          className="font-bold text-sm"
          htmlFor="Card Details"
        >
          Card Details
           
          <span
            className="text-midRed"
          >
            *
          </span>
        </label>
        <br />
        <input
          autoComplete="cc-number"
          className="mt-2 w-full ring-1 rounded-md py-2.5 px-3 outline-none ring-purp mb-8 text-textgray text-sm"
          id="Card Details"
          inputMode="numeric"
          maxLength={20}
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="w-full sm:w-1/2 flex flex-row items-stretch gap-x-12"
      >
        <div
          className="w-20"
        >
          <label
            className="font-bold text-sm"
            htmlFor="Expiry Date"
          >
            Expiry Date
             
            <span
              className="text-midRed"
            >
              *
            </span>
          </label>
          <br />
          <input
            className="mt-2 w-full ring-1 rounded-md py-2.5 px-3 outline-none ring-purp mb-8 text-textgray text-sm"
            id="Expiry Date"
            maxLength={5}
            placeholder="04/23"
            type="text"
          />
        </div>
        <div
          className="w-12"
        >
          <label
            className="font-bold text-sm"
            htmlFor="CVV"
          >
            CVV
             
            <span
              className="text-midRed"
            >
              *
            </span>
          </label>
          <br />
          <input
            className="mt-2 w-full ring-1 rounded-md py-2.5 px-3 outline-none ring-purp mb-8 text-textgray text-sm text-black "
            id="CVV"
            maxLength={3}
            placeholder="923"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="flex items-center gap-x-6 pb-24 mt-8"
    >
      <button
        className="text-white bg-gradient-to-r from-btnGradOne to-btnGradTwo py-2.5 rounded-lg px-16 font-bold text-sm "
        data-testid="test-btn"
        onClick={[Function]}
      >
        Next
      </button>
      <button
        className="text-purp pl-5 font-bold text-sm "
        data-testid="test-btn"
        onClick={[Function]}
      >
        Cancel Payment
      </button>
    </div>
  </form>
</div>
`;
